### ISSUE TITLE
configure issues with GMPL

### ISSUE THREAD

**Issue by svigerske** [coin-or/CoinUtils#13]
Issue created by migration from Trac.

Original creator: raderdavi

Original creation time: 2007-01-19 14:27:15

Assignee: somebody

Version: 

When I try to configure Cbc and I enable GMPL, the configuration script aborts with an error.  I am able to configure glpk for use with OsiGlpk, but cannot configure just GMPL in CoinUtils.  

email:  david.rader`@`rose-hulman.edu

**Comment by svigerske** [comment]
Attachment [config.log](https://www.coin-or.org/issue-attachments/CoinUtils/ticket13/config.log) by raderdavi created at 2007-01-19 14:27:44

config.log file

**Comment by svigerske** [comment]
Comment by ladanyi created at 2007-01-19 14:38:05

Replying to [ticket:13 raderdavi]:
> When I try to configure Cbc and I enable GMPL, the configuration script aborts with an error.  I am able to configure glpk for use with OsiGlpk, but cannot configure just GMPL in CoinUtils.  
> 
> email:  david.rader`@`rose-hulman.edu

Hi David,

I suspect this ticket should have been filed against BuildTools, but no big deal, I'll get in touch with the right people. In the meantime could you let reply in this ticket which version of cbc did you try to compile? A release? Something from svn (stable/trunk)? From your config.log I guess you got the Cbc-1.1.0 tarball, but please, confirm it if this is so.

Thanks,
--Laci

**Comment by svigerske** [comment]
Comment by ladanyi created at 2007-01-19 14:38:05

Changing assignee from somebody to ladanyi.

**Comment by svigerske** [comment]
Comment by ladanyi created at 2007-01-19 14:38:05

Changing status from new to assigned.

**Comment by svigerske** [comment]
Comment by raderdavi created at 2007-01-19 14:59:29

Sorry about not submitting to BuildTools.  

I downloaded the tarball Cbc-1.1.0.tgz

Thanks.

Dave

Replying to [comment:1 ladanyi]:
> Replying to [ticket:13 raderdavi]:
> > When I try to configure Cbc and I enable GMPL, the configuration script aborts with an error.  I am able to configure glpk for use with OsiGlpk, but cannot configure just GMPL in CoinUtils.  
> > 
> > email:  david.rader`@`rose-hulman.edu
> 
> Hi David,
> 
> I suspect this ticket should have been filed against BuildTools, but no big deal, I'll get in touch with the right people. In the meantime could you let reply in this ticket which version of cbc did you try to compile? A release? Something from svn (stable/trunk)? From your config.log I guess you got the Cbc-1.1.0 tarball, but please, confirm it if this is so.
> 
> Thanks,
> --Laci

**Comment by svigerske** [comment]
Comment by @LouHafer created at 2007-01-22 03:43:56

Not really a BuildTools problem, but a close relative.

Glpk plays a game with library names. The user sees mpl_initialize, but that's a #define. The symbol is really glp_mpl_initialize. In CoinUtils/configure.ac the call to AC_COIN_HAS_USER_LIBRARY needs to be tweaked, to
```
AC_COIN_HAS_USER_LIBRARY([Gmpl],[GMPL],[glpmpl.h],[glp_mpl_initialize])
```

Then, in CoinUtils/src/Makefile.am, look for AM_CPPFLAGS. It must be augmented with the include files for gmpl, as
```
if COIN_HAS_GMPL
  AM_CPPFLAGS += -I`$(CYGPATH_W) $(GMPLINCDIR)`
endif
```

Turns out this is already set up over in Osi, for glpk proper. Once I saw the problem, it was just a matter of adapting from Osi configure and Makefile.am files.

**Comment by svigerske** [comment]
Comment by ladanyi created at 2007-01-22 04:17:17

Fixed both in stable/1.0 and in trunk. The bug is important enough to warrant a new point release. I'll create one when there will be a new point release of BuildTools (fixing serious cygwin bugs) hopefully within a week.

Thanks Lou!

--Laci

**Comment by svigerske** [comment]
Comment by ladanyi created at 2007-01-22 04:17:17

Resolution: fixed
